src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+
+
+            try {
+                for(final Tuple2<RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID, RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence> contigInfo : contigsCollection.getContents()) {
+                    final String contigId = contigInfo._1.toString();
+                    final byte[] sequence = contigInfo._2.toString().getBytes();
+                    final AlnRgn[] alnRgns = bwaAlignSequence(bwaMem, contigId, sequence);
+
+                    // todo: parse cigar for internal indels
+                    if (alnRgns.length > 1) {
+
+                        // filter out secondary alignments, convert to AlignmentRegion objects and sort by alignment start pos
+                        final List<AlignmentRegion> alignmentRegionList = Arrays.stream(alnRgns)
+                                .filter(a -> a.getSecondary() == -1)
+                                .map(AlignmentRegion::new)
+                                .sorted(Comparator.comparing(a -> a.startInContig))
+                                .collect(Collectors.toList());
+
+                        AlignmentRegion current = alignmentRegionList.get(0);
+                        AlignmentRegion previous;
+                        for (int i = 0; i < alignmentRegionList.size(); i++) {
+                            if (i > 0) {
+                                previous = current;
+                                current = alignmentRegionList.get(i);
+                                final AssembledBreakpoint assembledBreakpoint = new AssembledBreakpoint();
+                                assembledBreakpoint.contigId = contigId;
+
+                                assembledBreakpoint.region1 = previous;
+                                assembledBreakpoint.region2 = current;
+
+                                assembledBreakpoint.homology = "NA";
+                                if (assembledBreakpoint.region1.endInContig >= assembledBreakpoint.region2.startInContig) {
+                                    assembledBreakpoint.homology = getSequence(contigsCollection, contigId, assembledBreakpoint.region2.startInContig - 1, assembledBreakpoint.region1.endInContig);
+                                }
+
+                                assembledBreakpoint.insertedSequence = "NA";
+                                if (assembledBreakpoint.region1.endInContig < assembledBreakpoint.region2.startInContig - 1) {
+                                    assembledBreakpoint.insertedSequence = getSequence(contigsCollection, contigId, assembledBreakpoint.region1.endInContig, assembledBreakpoint.region2.startInContig);
+                                }
+
+                                assembledBreakpoints.add(assembledBreakpoint);
+                            }
+                        }
+                    }
+                }
+        } catch (final IOException e) {
+            throw new GATKException("could not execute BWA");
```

Comment:
Since they're created in the contstructor and closed in the `close()` method, I think it makes sense to let this method throw an exception, and have the client be responsible for closing the resources (since it opened them explicitly). Otherwise if they do the right thing and call close later on, the resources will have already been destroyed. Does that make sense?